### PR TITLE
Assorted stub updates

### PR DESCRIPTION
**Pull request by bdarnell**
Fix some bugs in existing stubs and add new ones (mostly with stubgen).

The common thread among these modules is that they are used by Tornado.

### CODE REVIEW

**Comment by gvanrossum** [comment]
Thanks! I see nothing offensive there, so merging.

File: stdlib/2.7/inspect.pyi [hunk 1]
```diff
@@ -9,3 +9,12 @@ _FrameRecord = Tuple[_Frame, str, int, str, List[str], int]
 
 def currentframe() -> _FrameRecord: ...
 def stack(context: int = ...) -> List[_FrameRecord]: ...
+
+# namedtuple('ArgSpec', 'args varargs keywords defaults')
```
Comment (JukkaL) [inline stdlib/2.7/inspect.pyi]: `typing.NamedTuple` should work here and simplify the definition

File: stdlib/3/inspect.pyi [hunk 2]
```diff
@@ -31,4 +31,16 @@ class ArgSpec(tuple):
 
 def getargspec(func: object) -> ArgSpec: ...
 
+# namedtuple('FullArgSpec', 'args varargs varkw defaults kwonlyargs kwonlydefaults annotations')
```
Comment (JukkaL) [inline stdlib/3/inspect.pyi]: Here `typing.NamedTuple` would also work

Comment (gvanrossum) [inline stdlib/3/inspect.pyi]: OK, I applied the suggested changes using NamedTuple.

**Pull request merged**